scripts/Interface/reduction_gui/reduction/diffraction/diffraction_adv_setup_script.py

```diff
@@ -277,34 +285,42 @@ def from_xml(self, xml_str):
                                                                        default=AdvancedSetupScript.typeofcorrection)
 
             # Caching options
-            self.cache_dir = BaseScriptElement.getStringElement(
-                instrument_dom, 'cache_directory', default=self.__class__.cache_dir)
-
-            tempbool = BaseScriptElement.getStringElement(
-                instrument_dom, "clean_cache", default=str(int(self.__class__.clean_cache)))
+            # split it into the three cache dirs
+            # NOTE: there should only be three entries, if not, let it fail early
+            self.cache_dir1, self.cache_dir2, self.cache_dir3 = BaseScriptElement.getStringElement(
+                instrument_dom, 'cachedir', default=";;").split(";")
+
+            tempbool = BaseScriptElement.getStringElement(instrument_dom,
+                                                          "cleancache",
+                                                          default=str(
+                                                              int(self.__class__.clean_cache)))
             self.clean_cache = bool(int(tempbool))
 
     def reset(self):
         r"""reset instance's attributes with the values of the class' attributes
         """
-        class_attrs_selected = ['pushdatapositive',
-                                'unwrapref',
-                                'lowresref',
-                                'cropwavelengthmin',
-                                'cropwavelengthmax',
-                                'removepropmppulsewidth',
-                                'maxchunksize',
-                                'filterbadpulses',
-                                'bkgdsmoothpars',
-                                'stripvanadiumpeaks',
-                                'vanadiumfwhm',
-                                'vanadiumpeaktol',
-                                'vanadiumsmoothparams',
-                                'preserveevents',
-                                'extension',
-                                'outputfileprefix',
-                                # Caching options
-                                'cache_dir',
-                                'clean_cache']
+        class_attrs_selected = [
+            'pushdatapositive',
+            'unwrapref',
+            'lowresref',
+            'cropwavelengthmin',
+            'cropwavelengthmax',
+            'removepropmppulsewidth',
+            'maxchunksize',
+            'filterbadpulses',
+            'bkgdsmoothpars',
+            'stripvanadiumpeaks',
+            'vanadiumfwhm',
+            'vanadiumpeaktol',
+            'vanadiumsmoothparams',
+            'preserveevents',
+            'extension',
+            'outputfileprefix',
+            # Caching options
+            'cache_dir1',
```

Comment:
can these be more descriptive? Something like
```python
cache_dir_scan_save
cache_dir_scan_1
cache_dir_scan_2
```

Reply:
sure, I will update the name in the next batch of commits.